### PR TITLE
Refactor gradethis_setup() and .onLoad()

### LINKED ISSUE #206: Align .onLoad and gradethis_setup()
AFAICT `.onLoad()` does the same work as `gradethis_setup()`, but because some arguments of `gradethis_setup()` differ from the default options in `gradethis_default_options`, these two processes might be at odds with each other.

```r
library(gradethis)
getOption("gradethis.code.incorrect")
#> [1] "{pipe_warning()}{code_feedback()} {random_encouragement()}"
```

``` r
gradethis::gradethis_setup()
#> Loading required package: gradethis
getOption("gradethis.code.incorrect")
#> [1] "{ .message } { random_encouragement() }"
```

I think we should have `gradethis_setup()` operate more like `learnr::tutorial_options()`. So running

```r
library(gradethis)
gradethis_setup()
```

would be equivalent to

```r
library(gradethis)
```

Effectively, `gradethis_setup()` arguments would all default to `NULL`. The function would set non-`NULL` arguments or fill in unset arguments with the default values. Then `gradethis_default_options` becomes the source of truth about the default option values, but we would document all of the options and their meaning in `gradethis_setup()`.

@schloerke the one piece that's confusing me at the moment is why it's necessary for gradethis to set knitr options in `.onload()` rather than settings the `learnr::tutorial_options()` as in `gradethis_setup()`?

### PR DESCRIPTION
**Pull request by gadenbuie**
This PR makes gradethis's options consistent according to the following rules:

1. All options are prefixed with `gradethis.`
2. The option name matches the argument name in `gradethis_setup()`.
3. Options are documented in `gradethis_setup()` and should link to the functions where they have an affect

This means that I had to change a bunch of the option names slightly to bring the option names in line with the arguments of `gradethis_setup()`. This seemed preferable to changing the names in `gradethis_setup()` to align with options.

This PR also makes `gradethis_setup()` redundant. Now the default setup is done on package load. Calling `gradethis_setup()` always checks that all options are populated, but does nothing if the option already exists.

The flow looks like this:

1. On package load:
    * Call `gradethis_setup()` and ensure the gradethis defaults are set for learnr's knitr options
    * Set legacy options
2. Thereafter calling `gradethis_setup()`
    * Only explicitly sets the values included in the call to `gradethis_setup()`
    * Unless somehow an option hasn't been created or has been removed, then we populate it with the default value

Fixes #206

### CODE REVIEW

File: R/gradethis_setup.R [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  ...,
+  pass = NULL,
+  fail = NULL,
+  code_correct = NULL,
+  code_incorrect = NULL,
+  fail_code_feedback = NULL,
+  pipe_warning = NULL,
+  allow_partial_matching = NULL
+) {
+  set_opts <- as.list(match.call()[-1])
```
Comment (schloerke) [inline R/gradethis_setup.R:64]: Maybe we also add a message if no arguments are provided at all, that they can remove the function call?

**Review by schloerke — APPROVED** [review]

File: R/gradethis_setup.R (hunk 1 above)
Comment (gadenbuie) [reply R/gradethis_setup.R:64]: Hmmm yeah I was thinking this is like elevator door close buttons, it shouldn't hurt to have it but it doesn't do anything if everything loaded properly.

Comment (gadenbuie) [reply R/gradethis_setup.R:64]: ...but it makes people feel like they have a little more control :)

Comment (schloerke) [reply R/gradethis_setup.R:64]: Correct. Can leave as is as there is no harm in calling it with no parameters.  I guess if the examples are updated to remove calling `gradethis_setup()`, then I'm happy

Comment (gadenbuie) [reply R/gradethis_setup.R:64]: actually on second thought, maybe `gradethis_setup()` should _always_ set the default learnr options to make gradethis work. That way it always explicitly makes gradethis work, even if somehow the knitr options were messed up after `library(gradethis)` was called.

Comment (gadenbuie) [reply R/gradethis_setup.R:64]: or at least to always set the `exercise.checker` option

**Pull request merged**